here-naksha-lib-model/src/commonMain/kotlin/naksha/model/request/PropertyFilter.kt

```diff
@@ -0,0 +1,82 @@
+package naksha.model.request
+
+import naksha.base.AnyObject
+import naksha.base.Platform
+import naksha.jbon.JbFeatureDecoder
+import naksha.model.request.query.*
+
+class PropertyFilter(val req: ReadFeatures) : ResultFilter {
+
+    companion object {
+        const val PROPERTIES = "properties"
+    }
+
+    override fun call(a1: ResultTuple): ResultTuple? {
+        val pSearch = req.query.properties ?: return a1
+        if (a1.tuple == null) return null
+        if (a1.tuple!!.feature == null) return null
```

Comment:
Please avoid using `!!`, the compiler complains for a good reason, rather do this:
```kotlin
val feature = a1.tuple?.feature ?: return null
...
decoder.mapBytes(feature)
```
The code is cleaner, shorted, and less error prone.